### ISSUE TITLE
spike_functions neo type casting problem AS versus ASA

### ISSUE THREAD

**Issue by russelljjarvis** [scidash/neuronunit#83]
After updating the file neuronunit/capabilities/spike_functions.py to be consistent with the latest patches at scidash specifically on line 44

https://github.com/scidash/neuronunit/blob/dev/neuronunit/capabilities/spike_functions.py#L44

I am getting the error:

```
Score is Z = -2.31
Executing test InjectedCurrentAPWidthTest on model {'vpeak': '39.3339866631', 'v0': '-69.404909032', 'c': '-59.3027897137', 'b': '-4.67452242583e-09', 'vr': '-69.2347705173', 'a': '0.0958901253475', 'd': '0.198097793223', 'vt': '-43.047008566', 'C': '9.85809505867e-05', 'k': '0.000776185035677'}... {}
Traceback (most recent call last):
  File "nsga_parallel.py", line 572, in <module>
    fitnesses.append(evaluate_e(value,vmpop[key]))
  File "nsga_parallel.py", line 216, in evaluate_e
    assert get_neab.suite.judge(model)#passing in model, changes the model
  File "/opt/conda/lib/python3.5/site-packages/sciunit-0.1.5.8-py3.5.egg/sciunit/__init__.py", line 424, in judge
    stop_on_error,deep_error)
  File "/opt/conda/lib/python3.5/site-packages/sciunit-0.1.5.8-py3.5.egg/sciunit/__init__.py", line 456, in judge_one
    deep_error=deep_error)
  File "/opt/conda/lib/python3.5/site-packages/sciunit-0.1.5.8-py3.5.egg/sciunit/__init__.py", line 319, in judge
    raise score.score # An exception.  
  File "/opt/conda/lib/python3.5/site-packages/sciunit-0.1.5.8-py3.5.egg/sciunit/__init__.py", line 313, in judge
    score = self._judge(model, skip_incapable=skip_incapable)
  File "/opt/conda/lib/python3.5/site-packages/sciunit-0.1.5.8-py3.5.egg/sciunit/__init__.py", line 258, in _judge
    prediction = self.generate_prediction(model)
  File "/home/jovyan/mnt/neuronunit/neuronunit/optimization/../../neuronunit/tests/__init__.py", line 446, in generate_prediction
    return super(InjectedCurrentAPWidthTest,self).generate_prediction(model)
  File "/home/jovyan/mnt/neuronunit/neuronunit/optimization/../../neuronunit/tests/__init__.py", line 410, in generate_prediction
    widths = model.get_AP_widths()
  File "/home/jovyan/mnt/neuronunit/neuronunit/optimization/../../neuronunit/capabilities/__init__.py", line 82, in get_AP_widths
    action_potentials = self.get_APs()
  File "/home/jovyan/mnt/neuronunit/neuronunit/optimization/../../neuronunit/models/reduced.py", line 43, in get_APs
    waveforms = sf.get_spike_waveforms(vm)
  File "/home/jovyan/mnt/neuronunit/neuronunit/optimization/../../neuronunit/capabilities/spike_functions.py", line 43, in get_spike_waveforms
    snippets = [vm.time_slice(t-width/2,t+width/2) for t in spike_train]
  File "/home/jovyan/mnt/neuronunit/neuronunit/

```

I decided that the problem is vm is assumed to be of type neo AnalogueSignalArray as opposed to just regular AnalogueSignal (regular AnalogueSignal type lacks the time_slice method). 

I have inserted the line as follows at line 43/44 to make sure `vm` is cast from regular AS to ASArray

https://github.com/scidash/neuronunit/blob/dev/neuronunit/capabilities/spike_functions.py#L44
```
vm = neo.AnalogSignalArray(vm,
                                units=vm.units,
                                sampling_rate=vm.sampling_rate)
```

The test executes, but the score is insufficient data

```
Executing test RheobaseTest on model {'a': '0.11253334831', 'C': '9.47904374562e-05', 'vr': '-66.6179391146', 'd': '0.13829309117', 'vpeak': '39.2509799576', 'k': '0.00092769205166', 'c': '-56.5901791866', 'v0': '-53.9732204429', 'b': '-3.72973426155e-09', 'vt': '-36.2791070822'}... Score is Ratio = 1.65
Executing test InputResistanceTest on model {'a': '0.11253334831', 'C': '9.47904374562e-05', 'vr': '-66.6179391146', 'd': '0.13829309117', 'vpeak': '39.2509799576', 'k': '0.00092769205166', 'c': '-56.5901791866', 'v0': '-53.9732204429', 'b': '-3.72973426155e-09', 'vt': '-36.2791070822'}... {}
Score is Z = 1.03
Executing test TimeConstantTest on model {'a': '0.11253334831', 'C': '9.47904374562e-05', 'vr': '-66.6179391146', 'd': '0.13829309117', 'vpeak': '39.2509799576', 'k': '0.00092769205166', 'c': '-56.5901791866', 'v0': '-53.9732204429', 'b': '-3.72973426155e-09', 'vt': '-36.2791070822'}... {}
length of vm_fit 403 offset number 4
Score is Z = 4.83
Executing test CapacitanceTest on model {'a': '0.11253334831', 'C': '9.47904374562e-05', 'vr': '-66.6179391146', 'd': '0.13829309117', 'vpeak': '39.2509799576', 'k': '0.00092769205166', 'c': '-56.5901791866', 'v0': '-53.9732204429', 'b': '-3.72973426155e-09', 'vt': '-36.2791070822'}... {}
length of vm_fit 403 offset number 4
Score is Z = 30.31
Executing test RestingPotentialTest on model {'a': '0.11253334831', 'C': '9.47904374562e-05', 'vr': '-66.6179391146', 'd': '0.13829309117', 'vpeak': '39.2509799576', 'k': '0.00092769205166', 'c': '-56.5901791866', 'v0': '-53.9732204429', 'b': '-3.72973426155e-09', 'vt': '-36.2791070822'}... {}
Score is Z = -2.25
Executing test InjectedCurrentAPWidthTest on model {'a': '0.11253334831', 'C': '9.47904374562e-05', 'vr': '-66.6179391146', 'd': '0.13829309117', 'vpeak': '39.2509799576', 'k': '0.00092769205166', 'c': '-56.5901791866', 'v0': '-53.9732204429', 'b': '-3.72973426155e-09', 'vt': '-36.2791070822'}... {}
Score is Z = -1.46
Executing test InjectedCurrentAPAmplitudeTest on model {'a': '0.11253334831', 'C': '9.47904374562e-05', 'vr': '-66.6179391146', 'd': '0.13829309117', 'vpeak': '39.2509799576', 'k': '0.00092769205166', 'c': '-56.5901791866', 'v0': '-53.9732204429', 'b': '-3.72973426155e-09', 'vt': '-36.2791070822'}... {}
Score is Z = -1.55
Executing test InjectedCurrentAPThresholdTest on model {'a': '0.11253334831', 'C': '9.47904374562e-05', 'vr': '-66.6179391146', 'd': '0.13829309117', 'vpeak': '39.2509799576', 'k': '0.00092769205166', 'c': '-56.5901791866', 'v0': '-53.9732204429', 'b': '-3.72973426155e-09', 'vt': '-36.2791070822'}... {}
Score is Insufficient Data
Traceback (most recent call last):
Score is Insufficient Data
Traceback (most recent call last):
  File "nsga_parallel.py", line 572, in <module>
    fitnesses.append(evaluate_e(value,vmpop[key]))
  File "nsga_parallel.py", line 216, in evaluate_e
    assert get_neab.suite.judge(model)#passing in model, changes the model
  File "/opt/conda/lib/python3.5/site-packages/pandas/core/generic.py", line 917, in __nonzero__
    .format(self.__class__.__name__))
```
Which actually seems to be a problem with pandas.

**Comment by rgerkin** [comment]
@russelljjarvis 
Should be fixed with neo 0.5 changes that have been merged.  But I'm not sure if maybe the issue is deeper than that, so I'll leave it open pending your comments.

**Comment by russelljjarvis** [comment]
Alright I pulled the changes from dev concerning `spike_functions.py`, the it seems to be using the function squeeze a lot, I do not understand what squeeze does but I remember it from MATLAB.

I am about to check if the new spike_functions.py file this fixs the problem.

At first I tried to merge all of the changes from dev, but there are some substantial changes to backends NEURON that I suspect will take a while to debug.
